### PR TITLE
python3Packages.pydyf: fix failing tests

### LINKED ISSUE #172259: `python39Packages.pydyf` build failure on x86_64-linux as of `5a68a427`
## Issue description
Build of `python39Packages.pydyf` failed on x86_64-linux as of 5a68a4274d8ca3bbd0c455428a6d287daabbb6a6. This is currently breaking `python3Packages.wandb`.

```
error: builder for '/nix/store/im7zg3h4v509p0c1py9s2iq5w5ckdma3-python3.9-pydyf-0.1.2.drv' failed with exit code 1;
       last 10 log lines:
       > >                   assert 0, (
       >                         f'Pixel ({x}, {y}) in {name}: '
       >                         f'reference rgba{reference}, got rgba{value}')
       > E                   AssertionError: Pixel (0, 5) in test_text: reference rgba(255, 255, 255), got rgba(223, 223, 223)
       >
       > tests/__init__.py:69: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/test_pydyf.py::test_transform - AssertionError: Pixel (2, 4) in ...
       > FAILED tests/test_pydyf.py::test_text - AssertionError: Pixel (0, 5) in test_...
       > ========================= 2 failed, 20 passed in 1.81s =========================
       For full logs, run 'nix log /nix/store/im7zg3h4v509p0c1py9s2iq5w5ckdma3-python3.9-pydyf-0.1.2.drv'.
error: 1 dependencies of derivation '/nix/store/82hbk1p4885qbnp3mxwfzdxk8714p0fq-python3.9-weasyprint-54.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d96xzj6rsmj8dvwszkknq7l1aa6hb74j-python3.9-trytond-6.4.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/y4ir53icxj2mc8lbf67jn276k09r42li-python3.9-sentry-sdk-1.5.11.drv' failed to build
error: 1 dependencies of derivation '/nix/store/98dmx1xcnbj67nb4k27wgzhpw10jrfd9-python3.9-wandb-0.12.16.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2297522381)

python3Packages.wandb maintainers cc: @samuela
python39Packages.pydyf maintainers cc: @Prillan
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 5a68a4274d8ca3bbd0c455428a6d287daabbb6a6
2. Run `nix-build -A python39Packages.pydyf`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.0-1022-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0`
 - nixpkgs: `/nix/store/xjqhdxv0qzhx0pn3izd767rw6qnpv9b0-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2297522381) that created this issue.
- Internal tags: -9053734745789360131 1692909980607733186

### PR DESCRIPTION
**Pull request by Prillan**
###### Description of changes

Some tests were failing with the latest Ghostscript version. It's fixed upstream but no new version has been released, so for now we just apply that fix.

https://github.com/CourtBouillon/pydyf/commit/d4c34823f1d15368753c9c26f7acc7a24fc2d979

Fixes https://github.com/NixOS/nixpkgs/issues/172259 @samuela 

ZHF #172160 

Should be backported to release-22.05

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pydyf/default.nix [hunk 1]
```diff
@@ -19,6 +20,15 @@ buildPythonPackage rec {
     sha256 = "sha256-Hi9d5IF09QXeAlp9HnzwG73ZQiyoq5RReCvwDuF4YCw=";
   };
 
+  patches = [
+    # Fix tests for Ghostscript 9.56
+    # Remove after v0.1.3 has been released
+    (fetchpatch {
+      url = "https://github.com/CourtBouillon/pydyf/commit/d4c34823f1d15368753c9c26f7acc7a24fc2d979.patch";
+      sha256 = "sha256-Fx5I/8nRG6ULWiTV12WLeiJp9XPLm2E2QiNLy86MJB4=";
```
Comment (thiagokokada) [inline pkgs/development/python-modules/pydyf/default.nix:28]:
```
error: hash mismatch in fixed-output derivation '/nix/store/kmkbv2mzzwnr6ryjh7mhrvb9f0ivqahv-d4c34823f1d15368753c9c26f7acc7a24fc2d979.patch.drv':
         specified: sha256-Fx5I/8nRG6ULWiTV12WLeiJp9XPLm2E2QiNLy86MJB4=
            got:    sha256-2hHZW/q5CbStbpSJYbm3b23qKXANEb5jbPGQ83uHC+Q=
```

Comment (Prillan) [reply pkgs/development/python-modules/pydyf/default.nix]: That's really odd, I must have done something wrong when I tested it locally.. Anyway, I could reproduce it and have now updated the hash.

Thanks!

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 172295` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>30 packages failed to build:</summary>
  <ul>
    <li>alerta-server</li>
    <li>cura</li>
    <li>gns3-gui</li>
    <li>gns3-server</li>
    <li>home-assistant-component-tests.sentry</li>
    <li>moodle-dl</li>
    <li>moz-phab</li>
    <li>netbox</li>
    <li>octoprint</li>
    <li>openai (python39Packages.openai)</li>
    <li>python310Packages.dnachisel</li>
    <li>python310Packages.flametree</li>
    <li>python310Packages.trytond</li>
    <li>python310Packages.weasyprint</li>
    <li>python310Packages.xml2rfc</li>
    <li>python39Packages.django-rq</li>
    <li>python39Packages.dnachisel</li>
    <li>python39Packages.flametree</li>
    <li>python39Packages.sentry-sdk</li>
    <li>python39Packages.spacy</li>
    <li>python39Packages.spacy-loggers</li>
    <li>python39Packages.spacy-transformers</li>
    <li>python39Packages.textacy</li>
    <li>trytond (python39Packages.trytond)</li>
    <li>python39Packages.wandb</li>
    <li>python39Packages.weasyprint</li>
    <li>xml2rfc (python39Packages.xml2rfc)</li>
    <li>skytemple</li>
    <li>sourcehut.metasrht</li>
    <li>tribler</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pydyf</li>
    <li>python39Packages.pydyf</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
@ofborg eval

**Comment by samuela** [comment]
Thanks so much for your prompt attention to this @Prillan!

**Pull request merged**